backend/database.js

```diff
@@ -44,10 +47,42 @@ sbp('sbp/selectors/register', {
     }
     let { HEAD: currentHEAD } = latestHEADinfo
     let prefix = '['
+    if (MAX_EVENTS_AFTER) {
+      const circularList = createCircularList(MAX_EVENTS_AFTER, undefined)
+      while (currentHEAD !== hash) {
+        const entry = await sbp('chelonia/db/getEntry', currentHEAD)
+        currentHEAD = entry.message().previousHEAD
+        circularList.add(entry)
+      }
+      const entry = await sbp('chelonia/db/getEntry', currentHEAD)
+      circularList.add(entry)
```

Comment:
However, @corrideat had a very interesting idea for how to avoid having to fetch all of the messages only to discard most of them: we can create a doubly-linked list in effect, by storing additional keys in the database that point forward instead of backward.

For example, if we have the chain:

```
hash1 <- hash2 <- hash3 <- hash4
```

Then when we add `hash2`, we also create an entry that looks like this: `hash2-next` with the value `hash3`

This way, when calling `eventsAfter` with a `hashX`, it very simply look up `hashX-next` and pull the value there, and so on and so forth until it hits `MAX_EVENTS_AFTER`

Reply:
Yeah very interesting, should already be much faster than fetching every message, although we'd have to write these new entries to the database. So maybe not all contract chains need this database field, according to whether `eventsAfter` is going to be used often enough with them.

Reply:
@snowteamer While I agree not every contract might make much use of this feature, for simplicity's sake let's do it for every contract. In the future, if we need to, we can worry about optimizing it.

Reply:
@snowteamer I think it's easier to do it for all contracts as it's done now. Otherwise, you should probably also disable the `eventsAfter` API for contracts that don't have the `nextHEAD` record, as computing it from scratch is pretty expensive in a large contract.